src/main/js/components/need/apply-edit.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import { withStyles } from '@material-ui/core/styles';
+import { withSnackbar } from 'notistack';
+import { requiresLogin } from '../../util';
+import TextField from '@material-ui/core/TextField';
+import { changeApplicationStateForNeed} from '../../actions/need';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import Button from '@material-ui/core/Button';
+
+const styles = theme => ({
+    apply: {
+        minWidth: '105px',
+        width: 'calc(50% - 6px)',
+        margin: '3px',
+    },
+});
+
+@withStyles(styles)
+@withSnackbar
+class NeedApplyEditComponent extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            need: props.need,
+            updating: false,
+        };
+    }
+
+    handleFailure() {
+        this.props.enqueueSnackbar('Fehler beim Aktualisieren der Bewerbung', {
+            variant: 'error',
+        });
+    }
+
+    toggleApplicationStatus(){
+      this.setState({
```

Comment:
Hier tuts eigentlich auch this.setState({ updating: true })

Reply:
hast Recht